src/components/sky-nav/sky-nav.ts

```diff
@@ -13,55 +13,107 @@ import tokens from '../../compiled/tokens/js/tokens';
  */
 export const initSkyNav = (navButton: HTMLButtonElement) => {
   const menu = navButton.nextElementSibling as HTMLElement;
+  const navWrapper = navButton.closest('.c-sky-nav') as HTMLElement; // Review: should this use a .js- class or some different selector?
   const largeScreenMediaQuery = window.matchMedia(
     `(min-width: ${tokens.size.breakpoint.m.value})`
   );
+  const reducedMotionMediaQuery = window.matchMedia(
+    '(prefers-reduced-motion: reduce)'
+  );
 
   /**
    * Update Menu Layout
    * Sets visibility of menu & navButton for small vs large screen layouts.
    */
-  const update = (isLargeScreen: boolean) => {
+  const update = () => {
+    const isLargeScreen = largeScreenMediaQuery.matches;
     if (isLargeScreen) {
       navButton.removeAttribute('aria-expanded');
-      menu.removeAttribute('hidden');
+      menu.hidden = false;
     } else {
       navButton.setAttribute('aria-expanded', 'false');
       menu.hidden = true;
     }
   };
 
+  let timeoutId = -1;
+
   /**
    * Toggle Menu State (expanded/closed)
    * Sets aria-expanded & hidden attributes to show or hide the menu.
    */
   const toggle = () => {
-    const isExpanded =
-      navButton.getAttribute('aria-expanded') === 'true' || false;
+    const isExpanded = navButton.getAttribute('aria-expanded') === 'true';
 
     navButton.setAttribute('aria-expanded', String(!isExpanded));
-    menu.hidden = isExpanded;
-  };
 
-  /**
-   * Destroy Menu
-   * As part of the public API, allow users to remove the event listener
-   */
-  const destroy = () => navButton.removeEventListener('click', toggle);
+    if (reducedMotionMediaQuery.matches) {
+      menu.hidden = isExpanded;
+      return;
+    }
 
-  // Initialize the navButton with toggle behavior
-  navButton.addEventListener('click', toggle);
+    // We need to keep track of the siblings after the menu,
+    // because we will push them down for the animation
+    const elementsToShift: HTMLElement[] = [navWrapper];
+    let sibling: HTMLElement | null = navWrapper;
+    // eslint-disable-next-line no-unmodified-loop-condition
+    while ((sibling = sibling.nextElementSibling as HTMLElement | null)) {
+      elementsToShift.push(sibling);
+    }
+
+    const duration = 0.5;
```

Comment:
I copied this from https://github.com/cloudfour/cloudfour.com-patterns/blob/main/src/assets/toolkit/scripts/lib/component/sky.js#L7